src/diffusers/models/unet_2d_condition.py

```diff
@@ -285,122 +246,14 @@ def __init__(
         )
 
         # time
-        if time_embedding_type == "fourier":
-            time_embed_dim = time_embedding_dim or block_out_channels[0] * 2
-            if time_embed_dim % 2 != 0:
-                raise ValueError(f"`time_embed_dim` should be divisible by 2, but is {time_embed_dim}.")
-            self.time_proj = GaussianFourierProjection(
-                time_embed_dim // 2, set_W_to_weight=False, log=False, flip_sin_to_cos=flip_sin_to_cos
-            )
-            timestep_input_dim = time_embed_dim
-        elif time_embedding_type == "positional":
-            time_embed_dim = time_embedding_dim or block_out_channels[0] * 4
-
-            self.time_proj = Timesteps(block_out_channels[0], flip_sin_to_cos, freq_shift)
-            timestep_input_dim = block_out_channels[0]
-        else:
-            raise ValueError(
-                f"{time_embedding_type} does not exist. Please make sure to use one of `fourier` or `positional`."
-            )
-
-        self.time_embedding = TimestepEmbedding(
-            timestep_input_dim,
-            time_embed_dim,
-            act_fn=act_fn,
-            post_act_fn=timestep_post_act,
-            cond_proj_dim=time_cond_proj_dim,
-        )
-
-        if encoder_hid_dim_type is None and encoder_hid_dim is not None:
-            encoder_hid_dim_type = "text_proj"
-            self.register_to_config(encoder_hid_dim_type=encoder_hid_dim_type)
-            logger.info("encoder_hid_dim_type defaults to 'text_proj' as `encoder_hid_dim` is defined.")
+        time_embed_dim, timestep_input_dim = self._set_time_embed_layer(flip_sin_to_cos, freq_shift, block_out_channels, act_fn, time_embedding_type, time_embedding_dim, timestep_post_act, time_cond_proj_dim)
```

Comment:
Let's separate this function into `_set_time_proj` and `set_time_embedding` and make sure we have consistent naming logic across all these methods. i.e. 
* `_set_time_proj()` sets `self.time_proj` layer, 
* `_set_time_embedding()` set `self.time_embedding` layer,  
* `_set_encoder_hid_proj()` set `self.encoder_hid_proj` layer
* `_set_class_embedding()` sets `self.class_embedding` layer
* `_set_add_embedding()` sets `self.add_embedding` layer

IMO this is especially important for functions that update things in-place to help our users to have a good sense of what's been update without having to go into the code

Reply:
This also gives a nice chronology of the layers to the readers.

Reply:
Could it also make sense to add a `_` at the end of the function names to denote that the functions do things in-place?

Reply:
I think `_set` is a nice prefix to denote inplace operations. `_` at the end is a bit unconventional.

Reply:
- currently there is only one impl for time_embedding layer, so `_set_time_embedding` is not necessary?
- `_set` prefix also act as a indicator for internal use and should not be called directly by users.
- `_` at the end actually follows the pytorch naming style for in-place actions, but it might not be necessary if we already have a prefix?